src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,348 @@
+      };
+    }
+    if (inputStatus === "error") {
+      return {
+        borderColor: IOColors["error-600"],
+        borderWidth: 1
+      };
+    }
+    return {
+      borderColor: IOColors["grey-200"],
+      borderWidth: 1
+    };
+  }, [inputStatus]);
+
+  const animatedLabelProps = useAnimatedStyle(() => ({
+    fontSize: withTiming(interpolate(labelSharedValue.value, [0, 1], [16, 12])),
+    top: withTiming(interpolate(labelSharedValue.value, [0, 1], [0, -16]))
+  }));
+
+  useEffect(() => {
+    if (value.length > 0) {
+      labelSharedValue.value = 1;
+    } else {
+      if (inputStatus !== "focused") {
+        labelSharedValue.value = 0;
+      }
+    }
+  }, [labelSharedValue, value, inputStatus]);
+
+  const onTextInputPress = () => {
+    if (disabled) {
+      return;
+    }
+    labelSharedValue.value = 1;
+    setInputStatus("focused");
+    inputRef?.current?.focus();
+  };
+
+  const onChangeTextHandler = useCallback(
+    (text: string) => {
+      if (counterLimit && text.length > counterLimit) {
+        return;
+      }
+      onChangeText(text);
+    },
+    [counterLimit, onChangeText]
+  );
+
+  const onBlurHandler = useCallback(() => {
+    if (!value) {
+      labelSharedValue.value = 0;
+    }
+    if (secretInput) {
+      setSecretInput(true);
+    }
+    if (onValidate) {
+      const isValid = onValidate(value);
+      setIsValid(isValid);
+      if (!isValid) {
+        setInputStatus("error");
+        triggerHaptic("notificationError");
+        return;
+      } else {
+        triggerHaptic("notificationSuccess");
+      }
+    }
+    setInputStatus("initial");
+  }, [onValidate, value, labelSharedValue, secretInput]);
+
+  return (
+    <>
+      <Pressable
+        onPress={onTextInputPress}
+        style={[
+          inputStatus === "disabled" ? { opacity: 0.5 } : {},
+          boxStyle,
+          styles.textInput
+        ]}
+      >
+        {icon && (
+          <>
+            <Icon name={icon} size={24} />
+            <HSpacer size={8} />
+          </>
+        )}
+        <TextInput
+          editable={!disabled}
+          secureTextEntry={isSecretInput}
+          disableFullscreenUI={true}
+          blurOnSubmit={true}
+          ref={inputRef}
+          onFocus={() => {
+            setInputStatus("focused");
+            setIsValid(undefined);
+            labelSharedValue.value = 1;
+          }}
+          maxLength={counterLimit}
+          onBlur={onBlurHandler}
+          value={value}
+          onChangeText={onChangeTextHandler}
+          style={[
+            onValidate || secretInput ? { width: "80%" } : {},
+            styles.textInputStyle
+          ]}
+        />
+        <Animated.View
+          style={[styles.textInputLabelWrapper, icon ? { left: 32 } : {}]}
+        >
+          <Animated.Text
+            numberOfLines={1}
+            style={[
+              animatedLabelProps,
+              {
+                backgroundColor: IOColors.white,
```

Comment:
Have you encountered any particular problems with setting the `backgroundColor` property? Because I tried to remove it and the only change was the height of the input caret (which will change anyway once the floating label ends)